board/safety.h

```diff
@@ -504,16 +505,42 @@ bool steer_torque_cmd_checks(int desired_torque, int steer_req, const SteeringLi
     violation = true;
   }
 
-  // no torque if request bit isn't high
-  if ((steer_req == 0) && (desired_torque != 0)) {
-    violation = true;
+  bool steer_req_mismatch = (steer_req == 0) && (desired_torque != 0);
+
+  if (steer_req_mismatch) {
+    // no torque if request bit isn't high
+    if (!limits.has_steer_req_tolerance) {
+      violation = true;
+
+    // certain safety modes set their steer request bit low for one frame at a
+    // predefined max frequency to avoid steering faults in certain situations
+    } else {
+
+      // disallow torque cut if not enough recent matching steer_req messages
+      if (valid_steering_msg_count < (limits.min_valid_request_frames - 1)) {
+        violation = true;
+      }
+
+      // or we've cut torque too recently in time
+      uint32_t ts_elapsed = get_ts_elapsed(ts, ts_last_steer_req_mismatch);
+      if (ts_elapsed < limits.min_valid_request_rt) {
+        violation = true;
+      }
+
+      valid_steering_msg_count = 0;
```

Comment:
would call this `valid_steer_req_count` for consistency with `ts_last_steer_req_mismatch`